### PR TITLE
Upgrades packages.

### PR DESCRIPTION
**Pull request by sandro768**
----

**Note:** This PR includes upgrades to packages, and fixes for all the warnings/errors produced by updated eslint.

Ian decided that the strategy to be followed was to suppress these new warnings, to avoid combining dependency updates and extensive code changes in one PR.

The dependency update & suppression is going to happen in another PR, and we'll keep this one as a reference for what code fixes are required.

----

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have assigned myself to this PR.
- [x] I have chosen an appropriate label for the PR.
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.
- [x] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/client/tsconfig.json [hunk 1]
```diff
@@ -24,8 +24,9 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "preserve",
-    "composite": false
+    "jsx": "react-jsx",
+    "composite": false,
+    "noFallthroughCasesInSwitch": true
```
Comment (IanMayo) [inline packages/client/tsconfig.json:29]: @sandro768  - was there a reason for this switch-related change?  Personal preference?

Comment (sandro768) [reply packages/client/tsconfig.json:29]: No, I haven't changed this file.

My guess is that it was done automatically by yarn upgrade.

**Comment by IanMayo** [comment]
Thanks - I'll test it in a live app tomorrow.

Aah, hold on - I'll wait until you mark it as ready for review.

**Comment by sandro768** [comment]
I will mark after I test from my end. Currently can't make it due to #678

**Comment by IanMayo** [comment]
@sandro768  - I don't think you are blocked by #678 any more.  Can we progress this PR please?

**Comment by sandro768** [comment]
Sure, I am fixing TS and Lint errors.

**Pull request closed without merging**